translate/storage/l20n.py

```diff
@@ -52,10 +53,263 @@ class l20nfile(base.TranslationStore):
 
     def __init__(self, inputfile=None):
         super(l20nfile, self).__init__(unitclass=self.UnitClass)
-        pass
+        self.filename = getattr(inputfile, 'name', '')
+        if inputfile is not None:
+            l20nsrc = inputfile.read()
+            self.parse(l20nsrc)
+            self.makeindex()
 
     def parse(self, l20nsrc):
+        parser = L20nParser()
+        ast = parser.parse(l20nsrc)
+        print(ast)
         pass
 
     def __str__(self):
         pass
+
+class ParserError(Exception):
+    def __init__(self, message, pos, context):
+        self.name = 'ParserError'
+        self.message = message
+        self.pos = pos
+        self.context = context
+
+class L20nParser():
+    def __init__(self):
+        self._patterns = {
+            'identifier': re.compile(r'[A-Za-z_]\w*'),
+            'controlChars': re.compile(r'\\([\\\"\'{}])'),
+            'unicode': re.compile(r'\\u[0-9a-fA-F]{1,4}'),
+            'index': re.compile(r'@cldr\.plural\(\$?(\w+)\)'),
+            'placeables': re.compile(r'\{\{\s*([^\s]*?)\s*\}\}'),
+        }
+
+    def parse(self, string):
+        self._source = string
+        self._index = 0
+        self._length = len(string)
+
+        return self.getL20n()
+
+    def getAttributes(self):
```

Comment:
not sure about naming conventions in ttk - but should these be lower case methods?